### PR TITLE
Add Entry C++: Sets

### PR DESCRIPTION
**Pull request by KTom101**
New Entry on C++ Sets. From the shortlist; keyword 177.

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@KTom101 Thanks for submitting this PR! 😄

I left a few comments/suggestions for you to check out but wanted to make another recommendation here. I wonder if we could take the content under the "Inserting Values" and "Removing Values" sections and turn them into separate term entries called `insert.md`, `erase.md`, and `clear.md`?

File: content/cpp/concepts/sets/sets.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  - 'Game Development'
+Tags:
+  - 'Data Types'
+  - 'Sets'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+Sets are associative containers which store unique elements that can be referenced by an element's key. The keys are the values of the elements and they are constant, meaning they cannot be changed once assigned. However, existing elements can be removed or new elements can be added.
```
[the remark this answers is not in the document]
Comment (KTom101) [reply content/cpp/concepts/sets/sets.md:15]: @Dusch4593 I was reading in Bjarne Stroustrup's C++ book that you can say the keys are also the values in a set. I didn't really know which one I like to say better. But I think most sites just say "value". Maybe I can do something like the c++ reference site where they put in parenthesis that the value is the key. https://www.cplusplus.com/reference/set/set/

[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/cpp/concepts/sets/sets.md:15]: That sounds good! Thank you, Kyra! 🙏🏻

**Comment by KTom101** [comment]
@Dusch4593 I made changes based on your review and added some more.

File: content/cpp/concepts/sets/sets.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+std::set<datatype> setName;
+```
+
+A set can be created by using the `set` keyword and declaring a [data type](https://www.codecademy.com/resources/docs/cpp/data-types) and name.
+
+## Setting a Different Comparison Function
+
+By default, values of the set are sorted in ascending order. The comparison function can be changed from the default to `std::greater<int>` in order to sort the values in descending order:
+
+```psuedo
```
Comment (Dusch4593) [inline content/cpp/concepts/sets/sets.md:29]:
```suggestion
```pseudo
```

File: content/cpp/concepts/sets/terms/insert/insert.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Syntax
+
+```pseudo
+setName.insert(value);
+```
+
+- `value` is the element to be inserted; it must be of the same data type as the set it is being inserted into.
+
+## Example
```
Comment (Dusch4593) [inline content/cpp/concepts/sets/terms/insert/insert.md:25]: I recommend making this a Codebyte section since we can do it for C++ code.

**Review by Dusch4593 — APPROVED** [review]
Thanks @KTom101 Things LGTM for the next review but I left some super small comments/suggestions.

File: content/cpp/concepts/sets/terms/erase/erase.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  stoneFruits.erase("Apricots");
+  stoneFruits.erase("Nectarines");
+
+  // Print out set
+  std::set<std::string> :: iterator iter;
+  for(iter = stoneFruits.begin(); iter != stoneFruits.end(); iter++){
+    std::cout<< *iter << "\n";
+  }
+}
+```
```
Comment (SSwiniarski) [inline content/cpp/concepts/sets/terms/erase/erase.md:54]: You have an output example for `.insert()` but not here?

File: content/cpp/concepts/sets/sets.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+By default, values of the set are sorted in ascending order. The comparison function can be changed from the default to `std::greater<int>` in order to sort the values in descending order:
+
+```pseudo
+std::set<int, std::greater<int> > setName;
+```
+
+The following example shows this in use:
+
+```cpp
```
Comment (SSwiniarski) [inline content/cpp/concepts/sets/sets.md:35]: Maybe set an example for the default behavior, then show the change in output with the comparison function example?

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@KTom101, reviewed this and only had a couple of suggestions.

File: content/cpp/concepts/sets/terms/erase/erase.md (hunk 4 above)
Comment (KTom101) [reply content/cpp/concepts/sets/terms/erase/erase.md:54]: @SSwiniarski Yes, @Dusch4593 recommended I change the `.insert()` example to a codebyte example in the previous review and I forgot to remove it when making the changes. I'll take it out now, thanks!

File: content/cpp/concepts/sets/sets.md (hunk 5 above)
Comment (KTom101) [reply content/cpp/concepts/sets/sets.md:35]: @SSwiniarski I'll add that.

File: content/cpp/concepts/sets/sets.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  std::cout<< *iter << " ";
+  }
+}
+```
+
+This outputs the following:
+
+```shell
+10 19 25 42 
+```
```
Comment (KTom101) [inline content/cpp/concepts/sets/sets.md:56]:
```suggestion
```

```

File: content/cpp/concepts/sets/sets.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  {
+  std::cout<< *iter << " ";
+  }
+}
+```
+
+This outputs the following:
+
+```shell
+10 19 25 42
```
Comment (KTom101) [inline content/cpp/concepts/sets/sets.md:55]:
```suggestion
10 19 25 42 
```

```

**Comment by KTom101** [comment]
@SSwiniarski I've made the changes including adding an example with the default behavior. Let me know if you want to change anything around.

**Comment by KTom101** [comment]
@SSwiniarski Were the changes for this ok with you?

**Review by SSwiniarski — APPROVED** [review]
LGTM! 👍

**Comment by KTom101** [comment]
@Dusch4593 are there any final changes you want to make?

**Comment by Dusch4593** [comment]
@KTom101 Just made a few small changes in d7d9d6074214d61447c98d03fce7a4a50bc94b6e before merge.

- Small edits in a few "Syntax" sections.
- In 2 files, I changed "Example" sections to "Codebyte Example" and updated the snippets.

**Pull request merged**